src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,774 @@
+        #      "sourceAmount": "20.0",  // Fiat trade amount
+        #      "fiatCurrency": "EUR",   // Fiat token
+        #      "obtainAmount": "4.462", // Crypto trade amount
+        #      "cryptoCurrency": "LUNA",  // Crypto token
+        #      "totalFee": "0.2",    // Trade fee
+        #      "price": "4.437472",
+        #      "status": "Failed",  // Processing, Completed, Failed, Refunded
+        #      "createTime": 1624529919000,
+        #      "updateTime": 1624529919000
+        #   }
+        #   ],
+        #   "total": 1,
+        #   "success": true
+        # }
+        if _DATA in fiat_payments:
+            for payment in fiat_payments[_DATA]:
+                if payment[_STATUS] == "Completed":
+                    payment[_ISFIATPAYMENT] = True
+                    self._process_buy(payment, in_transactions, out_transactions)
+
+        # Process crypto deposits (limited to 90 day windows), fetches 1000 transactions
+        while current_start < now_time:
+            # The CCXT function only retrieves fiat deposits if you provide a valid 'legalMoney' code as variable.
+            crypto_deposits = self.client.fetch_deposits(params=({_STARTTIME: current_start, _ENDTIME: current_end}))
+
+            # CCXT returns a standardized response from fetch_deposits. 'info' is the exchange-specific information
+            # in this case from Binance.com
+
+            # {
+            #   'info': {
+            #       'amount': '0.00999800',
+            #       'coin': 'PAXG',
+            #       'network': 'ETH',
+            #       'status': '1',
+            #       'address': '0x788cabe9236ce061e5a892e1a59395a81fc8d62c',
+            #       'addressTag': '',
+            #       'txId': '0xaad4654a3234aa6118af9b4b335f5ae81c360b2394721c019b5d1e75328b09f3',
+            #       'insertTime': '1599621997000',
+            #       'transferType': '0',
+            #       'confirmTimes': '12/12',
+            #       'unlockConfirm': '12/12',
+            #       'walletType': '0'
+            #   },
+            #   'id': None,
+            #   'txid': '0xaad4654a3234aa6118af9b4b335f5ae81c360b2394721c019b5d1e75328b09f3',
+            #   'timestamp': 1599621997000,
+            #   'datetime': '2020-09-09T03:26:37.000Z',
+            #   'network': 'ETH',
+            #   'address': '0x788cabe9236ce061e5a892e1a59395a81fc8d62c',
+            #   'addressTo': '0x788cabe9236ce061e5a892e1a59395a81fc8d62c',
+            #   'addressFrom': None,
+            #   'tag': None,
+            #   'tagTo': None,
+            #   'tagFrom': None,
+            #   'type': 'deposit',
+            #   'amount': 0.00999800,
+            #   'currency': 'PAXG',
+            #   'status': 'ok',
+            #   'updated': None,
+            #   'internal': False,
+            #   'fee': None
+            # }
+
+            for deposit in crypto_deposits:
+                self._process_transfer(deposit, intra_transactions)

```

Comment:
I see you have a debug log inside _process_transfer: perhaps you could handle all debug logs at this level, so that lower-level functions don't have to worry about it, but you don't have to do it, it's just a suggestion.  The important thing is to capture all native data in the debug log.